packages/core/lib/test/async-every.test.ts

```diff
@@ -1,12 +1,10 @@
-const { describe, it, expect } = global
-
-const every = require('../async-every')
+import every from '../async-every'
 
 describe('async-every', () => {
   it('handles iterator errors', done => {
-    every([1], (item, cb) => cb(new Error('derp')), (err, result) => {
-      expect(err)
-      expect(err.message).toBe('derp')
+    every([1], (item, cb) => cb(new Error('derp')), (err) => {
+      expect(err).toBeTruthy()
+      expect(err!.message).toBe('derp')
```

Comment:
This non-null assertion operator is currently triggering a lint warning:

```
> eslint --ext .ts,.js--report-unused-disable-directives .


/Users/dan/bugsnag-js/packages/core/lib/test/async-every.test.ts
  7:14  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion

✖ 1 problem (0 errors, 1 warning)
```

I think we should avoid letting warnings creep in with `--max-warnings=0` and decided whether this rule is globally disabled in test files or ignored on a per-usage basis.

Reply:
> I think we should avoid letting warnings creep in with --max-warnings=0

I'm happy with that rule :+1:

What do you think about this lint rule? It seems somewhat overzealous for tests. The trouble is the compiler doesn't know `.toBeTruthy()` will bail when `err` is `null` so it needs convincing. Is there a better construct to use on the `err` in this test?

The refactor seems pretty straightforward otherwise:

```ts
expect(err).toBeTruthy()
if (err) expect(err!.message).toBe('derp')
```